tests/api_workflow/test_api_workflow_datasources.py

```diff
@@ -17,351 +18,591 @@
+            side_effect=[response]
+        )
+        assert list(
+            client.download_raw_metadata_iter(
+                run_id="run-id",
+                relevant_filenames_artifact_id="relevant-filenames",
+            )
+        ) == [
+            ("file1", "url1"),
+            ("file2", "url2"),
+        ]
+        client._datasources_api.get_list_of_raw_samples_metadata_from_datasource_by_dataset_id.assert_called_once_with(
+            dataset_id="dataset-id",
+            var_from=0,
+            to=mocker.ANY,
+            relevant_filenames_run_id="run-id",
+            relevant_filenames_artifact_id="relevant-filenames",
+            use_redirected_read_url=False,
+        )
 
-def test_download_new_raw_samples__from_beginning(mocker: MockerFixture) -> None:
-    mocker.patch.object(ApiWorkflowClient, "__init__", return_value=None)
-    mocker.patch.object(
-        ApiWorkflowClient, "get_processed_until_timestamp", return_value=0
-    )
-    current_time = 5
-    mocker.patch("time.time", return_value=current_time)
-    mocked_download = mocker.patch.object(ApiWorkflowClient, "download_raw_samples")
-    mocked_update_timestamp = mocker.patch.object(
-        ApiWorkflowClient, "update_processed_until_timestamp"
-    )
-    client = ApiWorkflowClient()
-    client.download_new_raw_samples()
-    mocked_download.assert_called_once_with(
-        from_=0,
-        to=current_time,
-        relevant_filenames_file_name=None,
-        use_redirected_read_url=False,
-    )
-    mocked_update_timestamp.assert_called_once_with(timestamp=current_time)
-
-
-def test_download_raw_samples_predictions__relevant_filenames_artifact_id(
-    mocker: MockerFixture,
-) -> None:
-    mock_response = mocker.MagicMock()
-    mock_response.has_more = False
-    mocker.patch.object(ApiWorkflowClient, "__init__", return_value=None)
-    mocked_api = mocker.MagicMock()
-    mocked_method = mocker.MagicMock(return_value=mock_response)
-    mocked_api.get_list_of_raw_samples_predictions_from_datasource_by_dataset_id = (
-        mocked_method
-    )
-    client = ApiWorkflowClient()
-    client._dataset_id = "dataset-id"
-    client._datasources_api = mocked_api
-    client.download_raw_predictions(
-        task_name="task", run_id="foo", relevant_filenames_artifact_id="bar"
-    )
-    kwargs = mocked_method.call_args[1]
-    assert kwargs.get("relevant_filenames_run_id") == "foo"
-    assert kwargs.get("relevant_filenames_artifact_id") == "bar"
-
-    # should raise ValueError when only run_id is given
-    with pytest.raises(ValueError):
-        client.download_raw_predictions(task_name="foobar", run_id="foo")
-    # should raise ValueError when only relevant_filenames_artifact_id is given
-    with pytest.raises(ValueError):
-        client.download_raw_predictions(
-            task_name="foobar", relevant_filenames_artifact_id="bar"
-        )
-
-
-def test_download_raw_samples_metadata__relevant_filenames_artifact_id(
```

Comment:
Moved to `test_download_raw_metadata_iter__relevant_filenames_artifact_id ` further up